### PR TITLE
Fixed the hang issue with each EOS frame.

### LINKED ISSUE #367: SVT-HEVC is blocked when flushing
In any app (e.g. SvtHevcEncApp, ffmpeg, gstreamer) that uses the SVT-HEVC library, If I flush without attempting to encode anything, it will be block. 

In case of gstreamer plugin, blocking occurs when receiving EB_BUFFERFLAG_EOS buffer after sending EB_BUFFERFLAG_EOS buffer. So I worked around to avoid blocking when flushing without encoding in gstreamer.(https://github.com/OpenVisualCloud/SVT-HEVC/pull/340/commits/8886d14a74a24f3446f0c8d9ccc97f7cfacfa903)
```
#0  0x00007ffff64fb6d6 in futex_abstimed_wait_cancelable (private=0, abstime=0x0, expected=0, futex_word=0x7fff991a6530) at ../sysdeps/unix/sysv/linux/futex-internal.h:205
#1  do_futex_wait (sem=sem@entry=0x7fff991a6530, abstime=0x0) at sem_waitcommon.c:111
#2  0x00007ffff64fb7c8 in __new_sem_wait_slow (sem=0x7fff991a6530, abstime=0x0) at sem_waitcommon.c:181
#3  0x00007ffff64fb839 in __new_sem_wait (sem=<optimized out>) at sem_wait.c:42
#4  0x00007ffff523a0e1 in EbBlockOnSemaphore (semaphoreHandle=<optimized out>) at /home/jin/Downloads/SVT-HEVC/Source/Lib/Codec/EbThreads.c:185
#5  0x00007ffff5239f0a in EbGetFullObject (fullFifoPtr=0x7fff991a6500, wrapperDblPtr=wrapperDblPtr@entry=0x7fffffff4e50) at /home/jin/Downloads/SVT-HEVC/Source/Lib/Codec/EbSystemResourceManager.c:688
#6  0x00007ffff520c2a5 in EbH265GetPacket (h265EncComponent=<optimized out>, pBuffer=0x7fffffff4e98, picSendDone=<optimized out>) at /home/jin/Downloads/SVT-HEVC/Source/Lib/Codec/EbEncHandle.c:3962
#7  0x00007ffff5c5d0c3 in gst_svthevc_enc_receive_frame (encoder=0x5555559848a0, got_packet=0x7fffffff4f10, send=1) at ../subprojects/gst-plugins-bad/ext/svthevcenc/gstsvthevcenc.c:1619
#8  0x00007ffff5c5d7af in gst_svthevc_enc_flush_frames (encoder=0x5555559848a0, send=1) at ../subprojects/gst-plugins-bad/ext/svthevcenc/gstsvthevcenc.c:1721
```

I created a gst-testcase to flush without attempting to encoding.(https://github.com/OpenVisualCloud/SVT-HEVC/pull/340/commits/87238453132701aa4a15d26a8c746ad82bb4f052) To create this situation in other apps, you need to modify your internal code to flush without encoding.

### PR DESCRIPTION
**Pull request by Austin-Hu**
When the application send fake frames (such as with EOS flag), the
encoder should make each frame go through the whole encoding kernel
stages (from ResourceCoordination to Packetization), otherwise
EbH265GetPacket() would be blocked in waiting for the buffer posted
by Packetization.

Signed-off-by: Austin Hu <austin.hu@intel.com>

Fixes #367 .

### CODE REVIEW

**Comment by Austin-Hu** [comment]
Fixed for Issue #367 .

**Comment by 1480c1** [comment]
> Fixed for Issue #367 .

Please add it to the main body of the pr in the form of `Fixes #367`

**Comment by Austin-Hu** [comment]
> > Fixed for Issue #367 .
> 
> Please add it to the main body of the pr in the form of `Fixes #367`

Yes, I forgot that...

**Comment by 1480c1** [comment]
You can edit it using the 3 dots on the top right

**Comment by Austin-Hu** [comment]
> You can edit it using the 3 dots on the top right

Done, thanks! :)

**Comment by 1480c1** [comment]
https://help.github.com/en/github/managing-your-work-on-github/closing-issues-using-keywords

It needs to be in a certain format for the auto closing

File: Source/Lib/Codec/EbResourceCoordinationProcess.c [hunk 1]
```diff
@@ -658,21 +658,25 @@ void* ResourceCoordinationKernel(void *inputPtr)
...
-			EbGetEmptyObject(
-				contextPtr->resourceCoordinationResultsOutputFifoPtr,
-				&outputWrapperPtr);
-			outputResultsPtr = (ResourceCoordinationResults_t*)outputWrapperPtr->objectPtr;
-			outputResultsPtr->pictureControlSetWrapperPtr = prevPictureControlSetWrapperPtr;
-
-			// Post the finished Results Object
-			EbPostFullObject(outputWrapperPtr);
-		}
+        if (((pictureControlSetPtr->pictureNumber > 0) && (prevPictureControlSetWrapperPtr != (EbObjectWrapper_t*)EB_NULL)) || endOfSequenceFlag) {
```
Comment (tianjunwork) [inline Source/Lib/Codec/EbResourceCoordinationProcess.c]: I suggest add comment to the new logic of endOfSequenceFlag. What do you think?

Comment (Austin-Hu) [reply Source/Lib/Codec/EbResourceCoordinationProcess.c]: Added in the 2nd commit of this PR.

**Pull request merged**